src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp

```diff
@@ -159,23 +203,10 @@ CommonDispatchData GatherKernelRef::SetDefault(const gather_params& params, cons
     auto out_layout = params.outputs[0].GetLayout();
     std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws;
 
-    if (out_layout == DataLayout::bfyx) {
-        dispatchData.gws = {output.X().v, output.Y().v, output.Feature().v * output.Batch().v};
-        dims_by_gws = {{Tensor::DataChannelName::X},
-                       {Tensor::DataChannelName::Y},
-                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
-    } else if (out_layout == DataLayout::bfzyx) {
-        dispatchData.gws = {output.X().v, output.Y().v * output.Z().v, output.Feature().v * output.Batch().v};
-        dims_by_gws = {{Tensor::DataChannelName::X},
-                       {Tensor::DataChannelName::Y, Tensor::DataChannelName::Z},
-                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
-    } else {
-        dispatchData.gws = {output.X().v * output.Y().v, output.Z().v * output.W().v, output.Feature().v * output.Batch().v};
-        dims_by_gws = {{Tensor::DataChannelName::X, Tensor::DataChannelName::Y},
-                       {Tensor::DataChannelName::Z, Tensor::DataChannelName::W},
-                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
-    }
-
+    dispatchData.gws = {output.X().v * output.Y().v, output.Z().v * output.W().v, output.Feature().v * output.Batch().v};
+    dims_by_gws = {{Tensor::DataChannelName::X, Tensor::DataChannelName::Y},
+                   {Tensor::DataChannelName::Z, Tensor::DataChannelName::W},
+                   {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
```

Comment:
Why did you change this logic? This will change performance in 4d data where Z=W=0.

Reply:
At this time, I did not know how to obtain the dimension of the block format. I saw that other kernels using such gws and I thought this method was concise. I understand that modular operation is wasted when in 4D. It seems it can be handled with get_default_format(format) function. I will fix it soon.

Reply:
Done.